R/derive_params_growth_age.R

```diff
@@ -0,0 +1,285 @@
+#'     start_date = BRTHDT,
+#'     end_date = VSDT,
+#'     out_unit = "months",
+#'     trunc_out = FALSE
+#'   ) %>%
+#'   mutate(
+#'     AGECUR = ifelse(AGECUR_D >= 365.25 * 2, AGECUR_M, AGECUR_D),
+#'     AGECURU = ifelse(AGECUR_D >= 365.25 * 2, CURU_M, CURU_D)
+#'   )
+#'
+#' #' metadata is in months
+#' cdc_meta_criteria <- admiralpeds::cdc_wtage %>%
+#'   mutate(
+#'     age_unit = "months",
+#'     SEX = ifelse(SEX == 1, "M", "F")
+#'   )
+#'
+#' #' metadata is in days
+#' who_meta_criteria <- bind_rows(
+#'   (admiralpeds::who_wt_for_age_boys %>%
+#'     mutate(
+#'       SEX = "M",
+#'       age_unit = "days"
+#'     )
+#'   ),
+#'   (admiralpeds::who_wt_for_age_girls %>%
+#'     mutate(
+#'       SEX = "F",
+#'       age_unit = "days"
+#'     )
+#'   )
+#' ) %>%
+#'   rename(AGE = Day)
+#'
+#' criteria <- bind_rows(
+#'   cdc_meta_criteria,
+#'   who_meta_criteria
+#' ) %>%
+#'   rename(AGEU = age_unit)
+#'
+#' derive_params_growth_age(
+#'   advs,
+#'   sex = SEX,
+#'   age = AGECUR,
+#'   age_unit = AGECURU,
+#'   meta_criteria = criteria,
+#'   parameter = VSTESTCD == "WEIGHT",
+#'   analysis_var = VSSTRESN,
+#'   set_values_to_sds = exprs(
+#'     PARAMCD = "WTASDS",
+#'     PARAM = "Weight-for-age z-score"
+#'   ),
+#'   set_values_to_pctl = exprs(
+#'     PARAMCD = "WTAPCTL",
+#'     PARAM = "Weight-for-age percentile"
+#'   )
+#' )
+derive_params_growth_age <- function(dataset,
+                                     sex,
+                                     age,
+                                     age_unit,
+                                     meta_criteria,
+                                     parameter,
+                                     analysis_var,
+                                     bmi_cdc_correction = FALSE,
+                                     set_values_to_sds = NULL,
+                                     set_values_to_pctl = NULL) {
+  # Apply assertions to each argument to ensure each object is appropriate class
+  sex <- assert_symbol(enexpr(sex))
+  age <- assert_symbol(enexpr(age))
+  age_unit <- assert_symbol(enexpr(age_unit))
+  analysis_var <- assert_symbol(enexpr(analysis_var))
+  assert_data_frame(dataset, required_vars = expr_c(sex, age, age_unit, analysis_var))
+
+  assert_data_frame(meta_criteria, required_vars = exprs(SEX, AGE, AGEU, L, M, S))
+  if (bmi_cdc_correction == TRUE) {
+    assert_data_frame(meta_criteria, required_vars = exprs(SEX, AGE, AGEU, L, M, S, P95, Sigma))
+  }
+
+  assert_expr(enexpr(parameter))
+  assert_varval_list(set_values_to_sds, optional = TRUE)
+  assert_varval_list(set_values_to_pctl, optional = TRUE)
+
+  if (is.null(set_values_to_sds) && is.null(set_values_to_pctl)) {
+    abort("One of `set_values_to_sds`/`set_values_to_pctl` has to be specified.")
+  }
+
+  # create a unified join naming convention, hard to figure out in by argument
+  dataset <- dataset %>%
+    mutate(
+      sex_join := {{ sex }},
+      ageu_join := {{ age_unit }}
```

Comment:
I think these variables should be created with [`get_new_tmp_var()`](https://pharmaverse.github.io/admiraldev/dev/reference/get_new_tmp_var.html) (or just use the specified variables).